internal/charmstore/store.go

```diff
@@ -1425,6 +1425,35 @@ func (store *Store) Search(sp SearchParams) (SearchResult, error) {
 	return result, nil
 }
 
+// List lists the store for the given ListParams.
+// It returns a ListResult containing the results of the list.
+func (store *Store) List(sp ListParams) (ListResult, error) {
+	query := store.DB.Entities().Find(sp.Filters)
+	if sp.sort == nil {
+		query = query.Sort("_id")
+	} else {
+		query = query.Sort(sp.sort...)
+	}
+	var entities []*mongodoc.Entity
+	err := query.All(&entities)
+	if err != nil {
+		return ListResult{}, errgo.Mask(err)
+	}
+	r := ListResult{
+		Total:      len(entities),
+		Results:    make([]*router.ResolvedURL, 0, len(entities)),
+	}
+	for _, entity := range entities {
+		id := &router.ResolvedURL{
+			URL: *entity.PreferredURL(true),
```

Comment:
In a ResolvedURL the URL field has to be the full name of the URL. not the preferred one. Also you'll need to check the development flag here too.